### ISSUE TITLE
Conflicting variables generated from mix of regular and default module exports

### ISSUE THREAD

**Issue by mislav** [rollup/rollup#2752]
- Rollup Version: 1.6.0
- earliest affected Rollup version: [1.2.0](https://github.com/rollup/rollup/releases/tag/v1.2.0)
- Operating System (or Browser): macOS 10.14.3
- Node Version: v10.8.0

### How Do We Reproduce?

Clone https://github.com/mislav/rollup-shadowing and run `npm it`.

Basically, there are two source files:

```js
//# src/index.js
import './foo'

//# src/foo.js
export function foo() {
  return "foo"
}

export default function() {
  return "default"
}
```

Note the mix of a normal export and a default export from the `foo.js` module.

### Expected Behavior

Generating a rollup bundle from `src/index.js` yields no conflicting variable/function names.

### Actual Behavior

The generated rollup bundle has two conflicting declarations of `function foo() { ... }` in the same scope:

```js
//# dist/index.js
function foo() {
  return "foo"
}

function foo() {
  return "default"
}
```

This was discovered by pulling in dependencies from `d3-geo` and `d3-shape` modules, both of which have sub-modules which mix normal exports with default exports.

/cc @lukastaegert https://github.com/rollup/rollup/pull/2689

**Comment by lukastaegert** [comment]
Minimal REPL version: https://rollupjs.org/repl?version=1.6.0&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmltcG9ydCUyMHglMkMlMjAlN0Jmb28lN0QlMjBmcm9tJTIwJy4lMkZmb28nJTVDbiU1Q25jb25zb2xlLmxvZyh4JTJDJTIwZm9vKSUzQiUyMiUyQyUyMmlzRW50cnklMjIlM0F0cnVlJTdEJTJDJTdCJTIybmFtZSUyMiUzQSUyMmZvby5qcyUyMiUyQyUyMmNvZGUlMjIlM0ElMjJleHBvcnQlMjBmdW5jdGlvbiUyMGZvbygpJTIwJTdCJTVDbiUyMCUyMHJldHVybiUyMCU1QyUyMmZvbyU1QyUyMiU1Q24lN0QlNUNuJTVDbmV4cG9ydCUyMGRlZmF1bHQlMjBmdW5jdGlvbiUyMGZvbygpJTIwJTdCJTVDbiUyMCUyMHJldHVybiUyMCU1QyUyMmRlZmF1bHQlNUMlMjIlNUNuJTdEJTIyJTJDJTIyaXNFbnRyeSUyMiUzQWZhbHNlJTdEJTVEJTJDJTIyb3B0aW9ucyUyMiUzQSU3QiUyMmZvcm1hdCUyMiUzQSUyMmNqcyUyMiUyQyUyMm5hbWUlMjIlM0ElMjJteUJ1bmRsZSUyMiUyQyUyMmFtZCUyMiUzQSU3QiUyMmlkJTIyJTNBJTIyJTIyJTdEJTJDJTIyZ2xvYmFscyUyMiUzQSU3QiU3RCU3RCUyQyUyMmV4YW1wbGUlMjIlM0FudWxsJTdE

The key here is that the default export has an id that matches the other export. Hope to look into this soon.

**Comment by lukastaegert** [comment]
It seems my previous example was wrong as this was just a basic parse error. The actual issue was that default exports were not always properly included when not tree-shaking. Fix at #2758
